### PR TITLE
Normative: Fix TimeZoneName issue in FormatDateTimePattern

### PR DESCRIPTION
**Pull request by FrankYFTang**
@sffc  Address https://github.com/tc39/ecma402/issues/515 also see https://github.com/tc39/proposal-intl-extend-timezonename/pull/9

### CODE REVIEW

**Review by ryzokuken — COMMENTED** [review]
Are you sure this is an editorial change? It seems like the value of `_fv_` could potentially be changed?

File: spec/datetimeformat.html [hunk 1]
```diff
@@ -371,6 +371,11 @@ <h1>FormatDateTimePattern ( _dateTimeFormat_, _patternParts_, _x_, _rangeFormatO
             1. Let _f_ be the value of _dateTimeFormat_'s internal slot whose name is the Internal Slot column of the matching row.
             1. Let _fv_ be a String value representing the day period of _tm_ in the form given by _f_; the String value depends upon the implementation and the effective locale of _dateTimeFormat_.
             1. Append a new Record { [[Type]]: _p_, [[Value]]: _fv_ } as the last element of the list _result_.
+          1. Else if _p_ is equal to *"timeZoneName"*, then
```
Comment (Ms2ger) [inline spec/datetimeformat.html:374]: The super-long line starting `Else if _f_ is *"narrow"*,` below has an `If _p_ is *"timeZoneName"*,` sentence that's dead code now; please remove it.

**Review by Ms2ger — APPROVED** [review]

**Review by anba — APPROVED** [review]

**Review by Ms2ger — APPROVED** [review]

**Review by ryzokuken — APPROVED** [review]

**Comment by ryzokuken** [comment]
2021-05-25: This PR got TC39-TG1 consensus.

**Comment by ryzokuken** [comment]
2021-06-03: This PR got TC39-TG2 consensus.

**Comment by sffc** [comment]
2021-06-03 TG2 consensus: https://github.com/tc39/ecma402/blob/master/meetings/notes-2021-06-03.md#normative-fix-timezonename-issue-in-formatdatetimepattern

**Comment by ryzokuken** [comment]
This PR has multiple positive reviews, no outstanding comments and TG1 and TG2 consensus. Merging.

**Pull request merged**